app/plex_tools.py

```diff
@@ -354,55 +431,55 @@ def delete_collection(data):
         data.delete()
         print("| Collection deleted")
 
-def alt_id_lookup(plex, value):
-    req = Request('{}{}'.format(plex.url, value.key))
+def new_movie_agent_get_tmdb(plex, movie):
+    req = Request('{}{}'.format(plex.url, movie.key))
     req.add_header('X-Plex-Token', plex.token)
     req.add_header('User-Agent', 'Mozilla/5.0')
     with urlopen(req) as response:
         contents = response.read()
     bs = BeautifulSoup(contents, 'lxml')
-    imdb_id = None
     tmdb_id = None
     for guid_tag in bs.find_all('guid'):
         agent = urlparse(guid_tag['id']).scheme
         guid = urlparse(guid_tag['id']).netloc
-        if agent == 'imdb':
-            imdb_id = guid
-        elif agent == 'tmdb':
+        if agent == 'tmdb':
             tmdb_id = guid
-    return imdb_id, tmdb_id
+            break
+    return tmdb_id
 
 def create_cache(config_path):
-    cache = os.path.join(os.path.dirname(config_path), 'cache.db')
-    connection = sqlite3.connect(cache)
+    cache = "{}.cache".format(os.path.splitext(config_path)[0])
     with sqlite3.connect(cache) as connection:
         connection.row_factory = sqlite3.Row
         cursor = connection.cursor()
         cursor.execute(''' SELECT count(name) FROM sqlite_master WHERE type='table' AND name='guids' ''')
         if cursor.fetchone()[0] != 1:
-            print("| Initializing cache database.".format(cache))
-            cursor.execute('CREATE TABLE IF NOT EXISTS guids (plex_guid TEXT PRIMARY KEY, imdb_id TEXT, tmdb_id TEXT)')
+            print("| Initializing cache database at {}".format(cache))
+            cursor.execute('CREATE TABLE IF NOT EXISTS guids (plex_guid TEXT PRIMARY KEY, id TEXT, updated TEXT)')
+        else:
+            print("| Using cache database at {}".format(cache))
 
-def query_cache(config_path, key, column):
-    cache = os.path.join(os.path.dirname(config_path), 'cache.db')
+def query_cache(config_path, key):
+    cache = "{}.cache".format(os.path.splitext(config_path)[0])
+    id_to_return = None
+    update = None
     with sqlite3.connect(cache) as connection:
         connection.row_factory = sqlite3.Row
         cursor = connection.cursor()
         cursor.execute("SELECT * FROM guids WHERE plex_guid = ?", (key, ))
         row = cursor.fetchone()
-        if row:
-            return row[column]
+        if row and row["id"]:
+            datetime_object = datetime.strptime(row["updated"], "%Y-%m-%d")
+            time_between_insertion = datetime.now() - datetime_object
+            id_to_return = int(row["id"])
+            update = True if time_between_insertion.days > 30 else False
+    return id_to_return, update
 
-def update_cache(config_path, plex_guid, **kwargs):
-    cache = os.path.join(os.path.dirname(config_path), 'cache.db')
+def update_cache(config_path, plex_guid, input_id):
+    updated_date = datetime.now() - timedelta(days=random.randint(1,10))
```

Comment:
Why randomize this?

Reply:
so that your entire library doesn't update the same day. This will spread out the updates so that over time it will only update a percentage each day of the month vs one day having the entire library update. I would change the amount of random to be based on how often the user updates